sys/socket.txt

```diff
@@ -415,3 +427,438 @@ dlci_add {
+}
+
+ethtool_ringparam_cmd_flags = ETHTOOL_GRINGPARAM, ETHTOOL_SRINGPARAM
+ethtool_ringparam {
+	cmd						flags[ethtool_ringparam_cmd_flags, int32]
+	rx_max_pending			int32
+	rx_mini_max_pending		int32
+	rx_jumbo_max_pending	int32
+	tx_max_pending			int32
+	rx_pending				int32
+	rx_mini_pending			int32
+	rx_jumbo_pending		int32
+	tx_pending				int32
+}
+
+ethtool_channels_cmd_flags = ETHTOOLGCHANNELS, ETHTOOL_SCHANNELS
+ethtool_channels {
+	cmd				flags[ethtool_channels_cmd_flags, int32]
+	max_rx			int32
+	max_tx			int32
+	max_other		int32
+	max_combined	int32
+	rx_count		int32
+	tx_count		int32
+	other_count		int32
+	combined_count	int32
+}
+
+ethtool_pauseparam_cmd_flags = ETHTOOL_GPAUSEPARAM, ETHTOOL_SPAUSEPARAM
+ethtool_pauseparam {
+	cmd			flags[ethtool_pauseparam_cmd_flags, int32]
+	autoneg		int32
+	rx_pause	int32
+	tx_pause	int32
+}
+
+ethtool_gstrings {
+	cmd				const[ETHTOOL_GSTRINGS, int32]
+	string_set		int32
+	len				int32
+	data			int8
+}
+
+ethtool_sset_info {
+	cmd				const[ETHTOOL_GSSET_INFO, int32]
+	reserved		int32
+	sset_mask		int64
+	data			int8
+}
+
+ethtool_test {
+	cmd			const[ETHTOOL_TEST, int32]
+	flags		int32
+	reserved	int32
+	len			int32
+	data 		int64
+}
+
+ethtool_stats {
+	cmd				const[ETHTOOL_GSTATS, int32]
+	n_stats			int32
+	data 			int64
+}
+
+ethtool_perm_addr {
+	cmd 		const[ETHTOOL_GPERMADDR, int32]
+	size		int32
+	data 		int8
+}
+
+# http://elixir.free-electrons.com/linux/latest/source/include/uapi/linux/ethtool.h#L1565
+traffic_flow_types = TCP_V4_FLOW, UDP_V4_FLOW, SCTP_V4_FLOW, AH_ESP_V4_FLOW, TCP_V6_FLOW, UDP_V6_FLOW, SCTP_V6_FLOW, AH_ESP_V6_FLOW, AH_V4_FLOW, ESP_V4_FLOW, AH_V6_FLOW, ESP_V6_FLOW, IPV4_USER_FLOW, IP_USER_FLOW, IPV6_USER_FLOW, IPV4_FLOW, IPV6_FLOW, ETHER_FLOW
+
+ethtool_rxnfc_cmd_flags = ETHTOOL_GRXFH, ETHTOOL_SRXFH, ETHTOOL_GRXRINGS, ETHTOOL_GRXCLSRLCNT, ETHTOOL_GRXCLSRULE, ETHTOOL_GRXCLSRLALL, ETHTOOL_SRXCLSRLDEL, ETHTOOL_SRXCLSRLINS
+ethtool_rxnfc {
+	cmd					flags[ethtool_rxnfc_cmd_flags, int32]
+	flow_type			flags[traffic_flow_types, int32]
+	data				int64
+	fs					ethtool_rx_flow_spec
+	rule_cnt			int32
+	rule_locs			ptr[inout, int32]
+}
+
+ethtool_rx_flow_spec {
+	flow_type		flags[traffic_flow_types, int32]
+	h_u				ethtool_flow_union
+	h_ext			ethtool_flow_ext
+	m_u				ethtool_flow_union
+	m_ext			ethtool_flow_ext
+	ring_cookie		int64
+	location		int32
+}
+
+ethtool_flow_union [
+	tcp_ip4_spec		ethtool_tcpip4_spec
+	udp_ip4_spec		ethtool_tcpip4_spec
+	sctp_ip4_spec		ethtool_tcpip4_spec
+	ah_ip4_spec			ethtool_ah_espip4_spec
+	esp_ip4_spec		ethtool_ah_espip4_spec
+	usr_ip4_spec		ethtool_usrip4_spec
+	tcp_ip6_spec		ethtool_tcpip6_spec
+	udp_ip6_spec		ethtool_tcpip6_spec
+	sctp_ip6_spec		ethtool_tcpip6_spec
+	ah_ip6_spec			ethtool_ah_espip6_spec
+	esp_ip6_spec		ethtool_ah_espip6_spec
+	usr_ip6_spec		ethtool_usrip6_spec
+	ether_spec			ethhdr
+	hdata				array[int8, 52]
+]
+
+ethtool_flow_ext {
+	padding 	array[int8, 2]
+	h_dest 		array[int8, ETH_ALEN]
+	vlan_etype	int16
```

Comment:
Use `be` suffix for all big endian integers: `vlan_etype int16be`.
Here and below.